jetty-http3/http3-qpack/src/test/java/org/eclipse/jetty/http3/qpack/SectionAcknowledgmentTest.java

```diff
@@ -0,0 +1,79 @@
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http3.qpack;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jetty.http3.qpack.internal.instruction.SectionAcknowledgmentInstruction;
+import org.eclipse.jetty.util.BufferUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.eclipse.jetty.http3.qpack.QpackTestUtil.encode;
+import static org.eclipse.jetty.http3.qpack.QpackTestUtil.toBuffer;
+import static org.eclipse.jetty.http3.qpack.QpackTestUtil.toMetaData;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SectionAcknowledgmentTest
+{
+    private static final int MAX_BLOCKED_STREAMS = 5;
+    private static final int MAX_HEADER_SIZE = 1024;
+
+    private QpackEncoder _encoder;
+    private QpackDecoder _decoder;
+    private TestDecoderHandler _decoderHandler;
+    private TestEncoderHandler _encoderHandler;
+
+    @BeforeEach
+    public void before()
+    {
+        _encoderHandler = new TestEncoderHandler();
+        _decoderHandler = new TestDecoderHandler();
+        _encoder = new QpackEncoder(_encoderHandler, MAX_BLOCKED_STREAMS);
+        _decoder = new QpackDecoder(_decoderHandler, MAX_HEADER_SIZE);
+    }
+
+    @Test
+    public void testSectionAcknowledgmentForZeroRequiredInsertCountOnDecoder() throws Exception
+    {
+        // Encode a header with only a value contained in the static table.
+        ByteBuffer buffer = encode(_encoder, 0, toMetaData("GET", "/", "http"));
+
+        // No instruction since no addition to table.
+        Instruction instruction = _encoderHandler.getInstruction();
+        assertNull(instruction);
+
+        // Decoding should generate no instruction.
+        _decoder.decode(0, buffer, _decoderHandler);
+        instruction = _decoderHandler.getInstruction();
+        assertNull(instruction);
+    }
+
+    @Test
+    public void testSectionAcknowledgmentForZeroRequiredInsertCountOnEncoder() throws Exception
+    {
+        // Encode a header with only a value contained in the static table.
+        ByteBuffer buffer = encode(_encoder, 0, toMetaData("GET", "/", "http"));
+        System.err.println(BufferUtil.toDetailString(buffer));
+
+        // Parsing a section ack instruction on the encoder when we are not expecting it should result in QPACK_DECODER_STREAM_ERROR.
+        SectionAcknowledgmentInstruction instruction = new SectionAcknowledgmentInstruction(0);
+        ByteBuffer instructionBuffer = toBuffer(instruction);
+        QpackException error = assertThrows(QpackException.class, () -> _encoder.parseInstructions(instructionBuffer));
```

Comment:
You want to test for `QpackException.StreamException` here to be sure it's not a SESSION error.

Reply:
It is a `SessionException` and thats what I would expect it to be. The rfc says it must be treated "as a connection error".

I have added this check to the test case.